### ISSUE TITLE
java.io.tmpdir property not respected in java.io.File.createTempFile?

### ISSUE THREAD

**Issue by borkdude** [oracle/graal#7731]
Repro:

``` java
public class Temp {
    public static void main(String[] args) throws java.io.IOException {
        System.out.println("env " + System.getenv("TMPDIR"));
        System.out.println("prop " + System.getProperty("java.io.tmpdir"));
        System.out.println("tmpfile " + java.io.File.createTempFile("clojure-lsp.", ".out"));
    }
}
```

Compile to native image.

I would expect `java.io.File.createTempFile` to be affected by either `TMPDIR`env var or `java.io.tmpdir` system property, but it isn't:

```
$ TMPDIR=foo ./temp
env foo
prop /var/folders/j9/xmjlcym958b1fr0npsp9msvh0000gn/T/
tmpfile /var/folders/j9/xmjlcym958b1fr0npsp9msvh0000gn/T/clojure-lsp.295997061503198015.out

$ TMPDIR=foo ./temp -Djava.io.tmpdir=foo
env foo
prop foo
tmpfile /var/folders/j9/xmjlcym958b1fr0npsp9msvh0000gn/T/clojure-lsp.10085851538447407567.out

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reaching out, we'll take a look into this shortly

**Comment by oubidar-Abderrahim** [comment]
Tracked internally on GR 51257

**Comment by borkdude** [comment]
This still happens with `Oracle GraalVM 23+37.1`

**Comment by borkdude** [comment]
Tested with `Java version: 25+11-LTS, vendor version: Oracle GraalVM 25-dev+11.1` which seems to fix this issue

**Comment by zakkak** [comment]
FTR the issue seems to be fixed in GraalVM for JDK 25 by https://github.com/oracle/graal/commit/03bc71dd4c236d2b7f29892c9106c90ca75d31a1#diff-e6462ab7fcea98cf6d96cc67964ce7db6055aebcd5a00554234426fc0019e4cdR284 in https://github.com/oracle/graal/pull/10613

**Comment by borkdude** [comment]
Thanks. It would be awesome if this was released in the March 18 release so we don't have to wait until September, but I guess it's not possible anymore.

**Comment by wirthi** [comment]
Yeah, unfortunately it is the other way round: it is a serious enough fix that we cannot just blindly backport it, we risk to affect way more code that we intend to. We merged it to master and assess the impact now on all our CI testing.

April CPU seems unlikely at this point, but we hope to bring at least a targeted fix for this particular problem to the June CPU.

**Comment by borkdude** [comment]
No June CPU fix for this one right?

**Comment by wirthi** [comment]
Unfortunately not, sorry. The respective engineer is on leave and I overlooked it.

As this is fixed in 25, do the early access builds of Java 25 help you? https://github.com/graalvm/oracle-graalvm-ea-builds/releases

**Comment by borkdude** [comment]
@wirthi Yeah, I can use an EA build... but am I risking other bugs with those? :)

**Comment by wirthi** [comment]
We have a pretty solid CI, so I have high trust in any (stable) build we provide. 

Especially for the 25 release branch: it has been in stabilization for a few months now, with only dedicated fixes being backported to it now. If it still has bugs (well, it certainly has), we most likely release with them in September.

**Comment by borkdude** [comment]
Can I install EA builds using "script friendly urls"? It seems that's not documented?

**Comment by wirthi** [comment]
as [SDKman can provide those ea builds](https://github.com/graalvm/oracle-graalvm-ea-builds?tab=readme-ov-file#install-with-sdkman) I guess it is feasible? 
`wget https://github.com/graalvm/oracle-graalvm-ea-builds/releases/download/jdk-25.0.0-ea.30/graalvm-jdk-25.0.0-ea.30_linux-x64_bin.tar.gz`
worked on my machine (was redirected by our company proxy but worked fine)

**Comment by borkdude** [comment]
I guess SDKMAN doesn't download them from `download.oracle.com`, that's what I was wondering about. Yes, it's feasible with some temporary CI script changes. Thanks!

**Comment by wirthi** [comment]
yeah they are on github, as linked above

**Comment by borkdude** [comment]
Seems to work now, thanks!
